### PR TITLE
migrate to vanniktech/gradle-maven-publish-plugin

### LINKED ISSUE #53: gradle-nexus-staging-plugin is deprecated
and is recommended to migrate to https://github.com/gradle-nexus/publish-plugin/.

or completely another plugin, such as https://github.com/vanniktech/gradle-maven-publish-plugin?

### PR DESCRIPTION
**Pull request by yshrsmz**
resolve #53 

I've locally tested without signing, and it seems to be working.
Is there any way to test publishing?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,7 +84,7 @@ Publishes SDK to Sonatype repository and releases to Maven Central.
 (Usually you don't need to publish manually because CI/CD workflow publishes automatically.)
 
 ```
-./gradlew :bucketeer:publishReleasePublicationToSonatypeRepository
```
Comment (yshrsmz) [inline README.md:88]: CI config should be updated after merge.

**Comment by cre8ivejp** [comment]
> Is there any way to test publishing?

If you upload it using the suffix `-SNAPSHOT`, you can test if the publishing process is working and delete it later using the snapshot task.

Or do you mean how to test it using the Bitrise?

https://central.sonatype.org/publish/publish-guide/#introduction
```
Once released/published, you will not be able to remove/update/modify that artifact

We provide the option to publish artifacts using the -SNAPSHOT suffix in case that you need to do any test on your publishing process, but once it is released there is no possibility to change it. Please check the [Can I change (modify, delete, update) a component on Central?](https://central.sonatype.org/faq/can-i-change-a-component/) FAQ for more details.
```

Delete SNAPSHOT Task
https://help.sonatype.com/repomanager3/nexus-repository-administration/tasks/maven-snapshot-tasks#MavenSNAPSHOTTasks-DeleteSNAPSHOTTask

File: README.md [hunk 2]
```diff
@@ -85,7 +82,7 @@ Publishes SDK to Sonatype repository and releases to Maven Central.
...
+./gradlew :bucketeer:publish --no-daemon --no-parallel -PmavenCentralUsername=$USERNAME -PmavenCentralPassword=$PASSWORD
```
Comment (yshrsmz) [inline README.md:85]: credential should now be passed like this or declared as environment variables.
I'm gonna change CI config once this gets merged.

https://github.com/vanniktech/gradle-maven-publish-plugin#where-to-upload-to

**Review by yshrsmz — COMMENTED** [review]
ok I've tested locally and confirmed that I can upload both snapshot and release artifacts.

**Review by cre8ivejp — APPROVED** [review]
Thank you!!

**Comment by yshrsmz** [comment]
thanks!

**Pull request merged**